server/pages/index.js

```diff
@@ -370,11 +371,18 @@ module.exports = function() {
 			}
 		} );
 
-	app.get( '/browsehappy', function( request, response ) {
-		response.status( 200 ).render( 'browsehappy.jade', {
-			urls: generateStaticUrls( request )
+	app.get( '/browsehappy', setUpRoute, function( req, res, next ) {
+		const wpcomRe = /^https?:\/\/[A-z_-]+\.wordpress\.com$/;
+		const primaryBlogUrl = get( req, 'context.user.primary_blog_url', '' );
+		const isWpcom = wpcomRe.test( primaryBlogUrl );
+		req.context = Object.assign( {}, req.context, {
```

Comment:
Style nit, I've adopted to using object deconstruction over `Object.assign` lately:

```js
req.context = {
	...req.context,
	template: 'browsehappy.jade',
	dashboardUrl: isWpcom
		? primaryBlogUrl + '/wp-admin'
		: 'https://dashboard.wordpress.com/wp-admin/'
}
```